api-lib/createNomineeMutations.ts

```diff
@@ -0,0 +1,119 @@
+import assert from 'assert';
+import { gql } from './Gql';
+
+export const getNomineeFromAddress = async (
+  address: string,
+  circleId: number
+) => {
+  const { nominees } = await gql.q('query')({
+    nominees: [
+      {
+        where: {
+          _and: [
+            {
+              address: { _eq: address },
+              circle_id: { _eq: circleId },
+              ended: { _eq: false },
+            },
+          ],
+        },
+      },
+      {
+        id: true,
+      },
+    ],
+  });
+
+  return nominees;
+};
+
+export const getUserFromAddress = async (address: string, circleId: number) => {
+  const { users } = await gql.q('query')({
+    users: [
+      {
+        where: {
+          _and: [
+            {
+              address: { _eq: address },
+              circle_id: { _eq: circleId },
+              deleted_at: { _is_null: true },
+            },
+          ],
+        },
+      },
+      {
+        id: true,
+      },
+    ],
+  });
+
+  return users;
+};
+
+export const getUserFromProfileIdWithCircle = async (
+  profileId: number,
+  circleId: number
+) => {
+  const { profiles_by_pk } = await gql.q('query')({
+    profiles_by_pk: [
+      {
+        id: profileId,
+      },
+      {
+        users: [
+          {
+            where: {
+              circle_id: { _eq: circleId },
+            },
+          },
+          {
+            circle: {
+              nomination_days_limit: true,
+              min_vouches: true,
+            },
+            id: true,
+          },
+        ],
+      },
+    ],
+  });
+  assert(profiles_by_pk, 'Profile cannot be found');
+  const user = profiles_by_pk.users.pop();
+  assert(user, `user for circle_id ${circleId} not found`);
+  return user;
+};
+
+export const insertNominee = async (
+  nominatedByUserId: number,
+  circleId: number,
+  address: string,
+  name: string,
+  description: string,
+  nominationDaysLimit: number,
+  vouchesRequired: number
+) => {
```

Comment:
should make this a single object param instead of multiple params like this, cleaner that way and lets you specify fields in any order.
```suggestion
export const insertNominee = async (params: {
  nominatedByUserId: number,
  circleId: number,
  address: string,
  name: string,
  description: string,
  nominationDaysLimit: number,
  vouchesRequired: number
}) => {
```